agent/engine/docker_task_engine.go

```diff
@@ -238,6 +241,37 @@ func NewDockerTaskEngine(cfg *config.Config,
 	return dockerTaskEngine
 }
 
+// Reconcile state of host resource manager with task status in managedTasks Slice
+// Done on agent restarts
+func (engine *DockerTaskEngine) reconcileHostResources() {
+	logger.Info("Reconciling host resources")
+	for _, task := range engine.state.AllTasks() {
+		taskStatus := task.GetKnownStatus()
+		resources := task.ToHostResources()
+
+		// Release stopped tasks host resources
+		// Call to release here for stopped tasks should always succeed
+		// Idempotent release call
+		if taskStatus.Terminal() {
+			err := engine.hostResourceManager.release(task.Arn, resources)
```

Comment:
when will this be useful?  Will it only ever be a no-op if the agent is always starting from zero?

Reply:
Not particularly useful right now, but this is a generalized implementation for keeping `HostResourceManager` in sync with engine. So this might find more uses in future, such as to keep periodic sync between engine and resource manager.